services:
  pangolin:
    image: "fosrl/pangolin:1.8.0"
    container_name: "pangolin"
    restart: "unless-stopped"
    volumes:
      - "${DOCKER_DATA_DIR:?error}/${COMPOSE_PROJECT_NAME:?error}/:/app/config:z"
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:3001/api/v1/"]
      interval: "3s"
      timeout: "3s"
      retries: 15
    networks:
      proxy:
    labels:
      traefik.enable: true
      traefik.http.routers.pangolin-next.entrypoints: "websecure-internal,websecure-external"
      traefik.http.routers.pangolin-next.rule: "Host(`pangolin.${TRAEFIK_DOMAINNAME_1:?error}` && !PathPrefix(`/api/v1`))"
      traefik.http.routers.pangolin-next.service: "pangolin-next"
      traefik.http.routers.pangolin-api.entrypoints: "websecure-internal,websecure-external"
      traefik.http.routers.pangolin-api.rule: "Host(`pangolin.${TRAEFIK_DOMAINNAME_1:?error}` && PathPrefix(`/api/v1`))"
      traefik.http.routers.pangolin-api.service: "pangolin-next"
      traefik.http.routers.pangolin-ws.entrypoints: "websecure-internal,websecure-external"
      traefik.http.routers.pangolin-ws.rule: "Host(`pangolin.${TRAEFIK_DOMAINNAME_1:?error}`)"
      traefik.http.routers.pangolin-ws.service: "pangolin-api"
      traefik.http.services.pangolin-next.service: "pangolin"
      traefik.http.services.pangolin-next.loadbalancer.server.port: 3002
      traefik.http.services.pangolin-api.service: "pangolin"
      traefik.http.services.pangolin-api.loadbalancer.server.port: 3000

networks:
  proxy:
    external: true
    name: "${TRAEFIK_NETWORK_NAME:-proxy}"